pkg/controllers/rollout/controller.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package rollout features a controller to do rollout.
+package rollout
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils/controller"
+)
+
+// Reconciler recompute the cluster resource binding.
+type Reconciler struct {
+	client.Client
+
+	Recorder record.EventRecorder
+}
+
+// Reconcile triggers a single binding reconcile round.
+func (r *Reconciler) Reconcile(ctx context.Context, key controller.QueueKey) (ctrl.Result, error) {
+	startTime := time.Now()
+	crpName, ok := key.(string)
+	if !ok {
+		err := fmt.Errorf("got a resource key %+v not of type namespaced key", key)
+		klog.ErrorS(err, "we have encountered a fatal error that can't be retried")
+		return ctrl.Result{}, controller.NewUnexpectedBehaviorError(err)
+	}
+	klog.V(2).InfoS("Start to rollout the bindings", "memberCluster", crpName)
```

Comment:
Hi Ryan! There is a type here; it should be `clusterResourcePlacement` rather than `memberCluster` I assume?